tests/components/isy994/test_config_flow.py

```diff
@@ -0,0 +1,302 @@
+            result["flow_id"], MOCK_USER_INPUT,
+        )
+
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_existing_config_entry(hass: HomeAssistantType):
+    """Test if config entry already exists."""
+    MockConfigEntry(domain=DOMAIN, unique_id=MOCK_UUID).add_to_hass(hass)
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {}
+
+    with patch(PATCH_CONFIGURATION) as mock_config_class, patch(
+        PATCH_CONNECTION
+    ) as mock_connection_class:
+        isy_conn = mock_connection_class.return_value
+        isy_conn.get_config.return_value = None
+        mock_config_class.return_value = MOCK_VALIDATED_RESPONSE
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], MOCK_USER_INPUT,
+        )
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_ABORT
+
+
+async def test_import_flow_some_fields(hass: HomeAssistantType) -> None:
+    """Test import config flow with just the basic fields."""
+    with patch(PATCH_CONFIGURATION) as mock_config_class, patch(
+        PATCH_CONNECTION
+    ) as mock_connection_class, patch(PATCH_ASYNC_SETUP, return_value=True), patch(
+        PATCH_ASYNC_SETUP_ENTRY, return_value=True,
+    ):
+        isy_conn = mock_connection_class.return_value
+        isy_conn.get_config.return_value = None
+        mock_config_class.return_value = MOCK_VALIDATED_RESPONSE
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=MOCK_IMPORT_BASIC_CONFIG,
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["data"][CONF_HOST] == f"http://{MOCK_HOSTNAME}"
+    assert result["data"][CONF_USERNAME] == MOCK_USERNAME
+    assert result["data"][CONF_PASSWORD] == MOCK_PASSWORD
+
+
+async def test_import_flow_with_https(hass: HomeAssistantType) -> None:
+    """Test import config with https."""
+
+    with patch(PATCH_CONFIGURATION) as mock_config_class, patch(
+        PATCH_CONNECTION
+    ) as mock_connection_class, patch(PATCH_ASYNC_SETUP, return_value=True), patch(
+        PATCH_ASYNC_SETUP_ENTRY, return_value=True,
+    ):
+        isy_conn = mock_connection_class.return_value
+        isy_conn.get_config.return_value = None
+        mock_config_class.return_value = MOCK_VALIDATED_RESPONSE
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=MOCK_IMPORT_WITH_SSL,
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["data"][CONF_HOST] == f"https://{MOCK_HOSTNAME}"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)